lib/gitlib.sh

```diff
@@ -201,3 +201,20 @@ gitlib::push_master () {
   logecho -n "Pushing$dryrun_flag master branch: "
   logrun -s git push$dryrun_flag origin master || return 1
 }
+
+##############################################################################
+# Ensure running with a synced repo.
+gitlib::repo_state () {
+  local output=$(git fetch --dry-run)
```

Comment:
maybe add `-C $(dirname $(readlink -ne $BASH_SOURCE))/..` to make sure we're checking the right repo (e.g. if `anago` is called not from `kubernetes/release`).